railgun/internal/diagnostics/server.go

```diff
@@ -61,9 +98,22 @@ func installHandlers(mux *http.ServeMux) {
 	mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
 }
 
+// installDumpHandlers adds the config dump webservice to the given mux.
+func installDumpHandlers(mux *http.ServeMux) {
+	mux.HandleFunc("/debug/config/successful", lastConfig(&successfulConfigDump))
```

Comment:
The MUX handler is expensive mechanism. The action is searching for some options.

Reply:
Can you clarify the ask/expected change here? Unclear exactly what's expensive here, but we only set this up once at startup in any case.

Reply:
@ccfishk I seems to me that we would need to see some substantiation of the assertion of cost here in order to evaluate any tradeoffs, as I'm unclear on the matter as well.

I think anything we would do here would be a micro-optimization at best: I ultimately think the value add of Mux for simple and clear code is worth what I expect to be occasional nanosecond costs for something that isn't in the critical path.

Reply:
We're also already creating one for the pprof routes, so I don't think simply adding new paths incurs much cost.

Docs suggest that the HTTP server always uses a ServeMux regardless--if you don't create and register a new one (as we do here) it still uses a DefaultServeMux. Adding handlers using those package functions rather than `mux.HandleFunc()` just adds handlers to the DefaultServeMux.

Reply:
Gotcha. Lets leave as it is atm.

Reply:
@ccfishk added the following comment, but it came in as an edit to my comment instead of a new comment, so reposting here:
```

Here is a definition "A ServeMux is essentially a HTTP request router (or multiplexor). " The nested dependencies are pretty deep. Here we just implemented several handlers, 
Another lightweight option(s) could be: 
http.HandleFunc("/config/a", handlerA)
http.HandleFunc("/config/b", handlerB)
```

Reply:
@ccfishk it seems like the main concern is the nested dependencies. I don't think this results in any unwarranted runtime costs, but even still @rainest's read of the docs seems to indicate we're using Mux regardless. I don't expect a problem and I am in favor of considering this resolved.